crates/extensions/aws_mapper_ext/src/converter.rs

```diff
@@ -41,47 +45,93 @@ impl AwsConverter {
         } else {
-            let default_timestamp = self.add_timestamp.then(|| self.clock.now());
-
-            // serialize with ThinEdgeJson for measurements, for alarms and events just add the timestamp
-            let payload = if input.topic.name.starts_with("tedge/measurements") {
-                let mut serializer = ThinEdgeJsonSerializer::new_with_timestamp(default_timestamp);
-                tedge_api::parser::parse_str(input.payload_str()?, &mut serializer)?;
-
-                serializer.into_string()?
-            } else if input.topic.name.starts_with("tedge/events")
-                || input.topic.name.starts_with("tedge/alarms")
-                || input.topic.name.starts_with("tedge/health")
-            {
-                let mut payload_json: Map<String, Value> =
-                    serde_json::from_slice(input.payload.as_bytes())?;
-
-                if let Some(timestamp) = default_timestamp {
-                    let timestamp = timestamp
-                        .format(&time::format_description::well_known::Rfc3339)?
-                        .as_str()
-                        .into();
-                    payload_json.entry("time").or_insert(timestamp);
-                }
-
-                serde_json::to_string(&payload_json)?
-            } else {
-                return Ok(vec![]);
-            };
+            match self.mqtt_schema.entity_channel_of(&input.topic) {
+                Ok((source, channel)) => self.try_convert_te_topics(source, channel, input),
+                Err(_) => self.try_convert_tedge_topics(input),
+            }
+        }
+    }
 
+    fn try_convert_tedge_topics(
+        &mut self,
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        // serialize with ThinEdgeJson for health, just add the timestamp
+        if input.topic.name.starts_with("tedge/health") {
+            let payload = self.with_time_stamp(input)?;
             let topic_suffix = match input.topic.name.split_once('/') {
                 Some((_, topic_suffix)) => topic_suffix,
                 None => return Ok(vec![]),
             };
-
             let out_topic = Topic::new(&format!("aws/td/{topic_suffix}"))?;
 
             let output = MqttMessage::new(&out_topic, payload);
             self.size_threshold.validate(&output)?;
             Ok(vec![(output)])
+        } else {
+            Ok(vec![])
+        }
+    }
+
+    fn try_convert_te_topics(
+        &mut self,
+        source: EntityTopicId,
+        channel: Channel,
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        match &channel {
+            Channel::Measurement { measurement_type } => {
+                self.convert_message(input, source, measurement_type)
+            }
+
+            Channel::Event { event_type } => self.convert_message(input, source, event_type),
+
+            Channel::Alarm { alarm_type } => self.convert_message(input, source, alarm_type),
```

Comment:
[Minor] As the action is the same for all branches, this can be simpler and more explicit.

```suggestion
            Channel::Measurement { mea_type } |
            Channel::Event { mea_type }  |
            Channel::Alarm { mea_type } => self.convert_message(input, source, mea_type ),
```

Reply:
> [Minor] As the action is the same for all branches, this can be simpler and more explicit.

This will not work, because they have to be `specific structure filed names` for each MEA struct as mentioned in the declaration of an MEA struct.